OpenRA.Mods.TS/Traits/Buildings/LowBridge.cs

```diff
@@ -0,0 +1,55 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.TS.Traits
+{
+	[Desc("Bridge actor that can't be passed underneath.")]
+	class LowBridgeInfo : ITraitInfo, Requires<BuildingInfo>, Requires<HealthInfo>
+	{
+		public readonly string TerrainType = "Road";
+
+		public object Create(ActorInitializer init) { return new LowBridge(init.Self, this); }
+	}
+
+	class LowBridge : INotifyCreated
+	{
+		readonly LowBridgeInfo info;
+
+		public LowBridge(Actor self, LowBridgeInfo info)
+		{
+			this.info = info;
+
+			var health = self.Trait<Health>();
+			health.RemoveOnDeath = false;
+		}
+
+		void INotifyCreated.Created(Actor self)
+		{
+			if (self.IsDead)
+				return;
+
+			var buildingInfo = self.Info.TraitInfo<BuildingInfo>();
+			var cells = FootprintUtils.PathableTiles(self.Info.Name, buildingInfo, self.Location);
+			var tileSet = self.World.Map.Rules.TileSet;
+			foreach (var cell in cells)
+				self.World.Map.CustomTerrain[cell] = tileSet.GetTerrainIndex(info.TerrainType);
```

Comment:
Couldn't you move the `tileSet.GetTerrainIndex(info.TerrainType);` out of the loop?
E.g.
```
var terrainIndex = tileSet.GetTerrainIndex(info.TerrainType);
foreach (var cell in cells)
	self.World.Map.CustomTerrain[cell] = terrainIndex;
```